processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -344,20 +344,21 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
         }
     }
 
-    private void pollOnce(Consumer<String, byte[]> consumer) {
+    private void pollOnce(Consumer<byte[], byte[]> consumer) {
         Timer timer = Utils.timer();
-        ConsumerRecords<String, byte[]> records = consumer.poll(POLL_TIMEOUT_MILLIS);
+        ConsumerRecords<byte[], byte[]> records = consumer.poll(POLL_TIMEOUT_MILLIS);
         metrics.consumerPollTime.record(timer.duration());
 
         timer = Utils.timer();
         records.forEach(record -> {
             TopicPartition tp = new TopicPartition(record.topic(), record.partition());
             PartitionContext context = contexts.get(tp);
             DeferredCompletion completion = context.registerOffset(record.offset());
+            String key = processors.keyDeserializer().deserialize(record.key());
```

Comment:
`Processors` contains stuff that becomes necessary only at just before `DecatonProcessor` begins to process tasks. That is, other parent parts of decaton stays agnostic about what it does for the given task which is yet represented as `byte[]`. 
In contract, the key in the current impl is expected to be read as `String` in not only the parts around  `DecatonProcessor` but also at nearly everywhere of decaton impl starting from `ProcessorSubscription` to downwards.
So I think `keyDeserializer` should be a properly of `ProcessorSubscription` itself rather than one of `Processors` (hence `ProcessorBuilder` either).